### PR TITLE
Fix updating dynamic objects in LightmapGI

### LINKED ISSUE #77167: LightmapGI: Dynamic objects don't update immediately after baking lightmaps or (re)loading a scene
### Godot version

4.0 dev

### System information

Windows 10

### Issue description

When dynamic objects are baked, they will not receive global illumination from lightmap captures until they are updated by means such as toggling their dynamic status or by moving them. This issue seems to stem from the pairing code, which is not being executed on the instances following the baking process.

### Steps to reproduce

1. Add objects to a scene with their global illumination mode set to dynamic.
2. Bake a lightmap.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by huwpascoe**
Fixes https://github.com/godotengine/godot/issues/77167

> **Minimal reproduction project:** [test_lightmapgi_dynamic_object_update.zip](https://github.com/user-attachments/files/17414285/test_lightmapgi_dynamic_object_update.zip)

image

### CODE REVIEW

**Review by clayjohn — COMMENTED** [review]
Great work so far. I think the approach right now is a bit hacky so I have left some comments about my concerns and how to address them.

File: servers/rendering/renderer_scene_cull.cpp [hunk 1]
```diff
@@ -1132,6 +1132,16 @@ void RendererSceneCull::instance_set_visible(RID p_instance, bool p_visible) {
 		_unpair_instance(instance);
 	}
 
+	if (instance->base_type == RS::INSTANCE_LIGHTMAP) {
+		if (instance->scenario && p_visible) {
+			for (SelfList<Instance> *i = instance->scenario->instances.first(); i != nullptr; i = i->next()) {
```
[the remark this answers is not in the document]
Comment (huwpascoe) [reply servers/rendering/renderer_scene_cull.cpp:1137]: The unpair code is literally the line above : D

[the remark this answers is not in the document]
Comment (huwpascoe) [reply servers/rendering/renderer_scene_cull.cpp:1137]: > 2. It is pairing with every single object even though it only needs to pair with dynamic objects

The objects are checked if they are dynamic inside the pair function

**Comment by huwpascoe** [comment]
Made the changes. It works for all three scenarios: scene load, visibility change, rebake.

**Comment by huwpascoe** [comment]
Works with the new MRP https://github.com/godotengine/godot/issues/77167#issuecomment-2419764911

**Review by clayjohn — APPROVED** [review]
Looks great! Thank you for making the changes we discussed

**Comment by Repiteo** [comment]
Thanks! Congratulations on your first contribution!

**Comment by atirut-w** [comment]
Is this getting cherry picked?

**Comment by clayjohn** [comment]
> Is this getting cherry picked?

Good reminder. It wasn't tagged to be cherry picked. But I think it can be

**Pull request merged**